model/mustache.go

```diff
@@ -0,0 +1,79 @@
+package model
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/hpcloud/fissile/mustache"
+)
+
+// GetVariablesForRole returns all the environment variables required for
+// calculating all the templates for the role
+func (r *Role) GetVariablesForRole() ([]*ConfigurationVariable, error) {
```

Comment:
Not sure what you mean about pattern (A pattern of how we use Go?). I asked because I saw  (roles.go, line 123) the definition of `type ConfigurationVariableSlice []*ConfigurationVariable` and wondered if it would make sense to use the type name instead of replicating the definition